plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/error_resolution/CustomMarkerResolutionGenerator.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.error_resolution;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarker;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerFBNElement;
+import org.eclipse.ui.IMarkerResolution;
+import org.eclipse.ui.IMarkerResolutionGenerator2;
+
+public class CustomMarkerResolutionGenerator implements IMarkerResolutionGenerator2 {
+
+	@Override
+	public IMarkerResolution[] getResolutions(final IMarker marker) {
+		try {
+			final EObject obj = FordiacErrorMarker.getTarget(marker);
+			if (obj instanceof final ErrorMarkerFBNElement errorType) {
+				return new IMarkerResolution[] { new CreateMissingFBMarkerResolution(errorType),
+						new ChangeFBMarkerResolution(errorType) };
+			}
+		} catch (final IllegalArgumentException | CoreException e) {
+			e.printStackTrace();
+		}
+
+		return new IMarkerResolution[0];
+	}
+
+	@Override
+	public boolean hasResolutions(final IMarker marker) {
+		try {
+			final EObject target = FordiacErrorMarker.getTarget(marker);
+			return target instanceof ErrorMarkerFBNElement;
```

Comment:
This will potentially load all files with markers in it when updating the *Problems* view. I would again suggest to rely on the target type attribute.
```suggestion
			final EClass targetType = FordiacErrorMarker.getTargetType(marker);
			return LibraryElementPackage.Literals.ERROR_MARKER_FBN_ELEMENT.equals(targetType);
```